lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm

```diff
@@ -187,6 +188,40 @@ sub _validate_common_properties ($self) {
     $validation->optional('description');
 }
 
+sub _check_keep_logs_and_results ($self, $properties) {
+
+# TODO: try to understand why this throw `Mojo::Reactor::Poll: I/O watcher failed: A response has already been rendered`
```

Comment:
The return value must indicate to the calling method if it should return itself, so we need to explicitly return something true or false. I currently wouldn't know by heart if `render` would return true or false, so I think an explicit return statement is better

Reply:
Repeating myself:
The return value must indicate to the calling method if it should return itself, so we need to explicitly return something true or false.
The method is called like this:
```
return undef unless $self->_check_keep_logs_and_results($properties);
```
So let's assume, `render` returns true.
Now that would signal the calling method that the check passed. So it doesn't return itself, but goes on.
`render` is called a second time, which leads to the error message.

That's why I suggested to explicitly return 1 or 0.